### PR TITLE
Datasource context

### PR DESCRIPTION
**Pull request by phated**
I'd like to implement datasource for my structs, but my database connection pool is attached to the context. If we pass the context to `query_operation`, I should be able to implement this.

I also updated the book for Datasource based on my newer changes.

### CODE REVIEW

File: docs/zh-CN/src/cursor_connections.md [hunk 1]
```diff
@@ -22,25 +22,37 @@ impl DataSource for Integers {
...
+            QueryOperation::LastBefore {before, limit} => {
                 // 结束数字，如果没有before参数，则为0
-                let end = before
-                    .and_then(|before| before.parse::<i32>().ok())
+                let end = before.parse::<i32>()
+                    .ok()
                     .unwrap_or(0);
                 (end - *limit, end)
             }
+            // TODO: Advise to handle all cases
```
Comment (phated) [inline docs/zh-CN/src/cursor_connections.md:54]: @sunli829 this needs to be translated to chinese.

**Comment by phated** [comment]
@sunli829 should the book be deployed on Pull Request? I think that should only be Push to master.

**Comment by sunli829** [comment]
Thank you.😁

**Comment by sunli829** [comment]
> @sunli829 should the book be deployed on Pull Request? I think that should only be Push to master.

Let me change it. 😁

**Pull request merged**